File: bench/micro/vec_amortized.go
// Go equivalent of vec_amortized.ap.
// Build + consume + GC, single timed region.
package main

import (
	"fmt"
	"time"
)

func main() {
	n := 200000
	t0 := time.Now()

	v := make([]int, 0)
	for i := 0; i < n; i++ {
		v = append(v, i)
	}
	sum := 0
	for j := 0; j < n; j++ {
		sum += v[j]
	}

	elapsed := time.Since(t0).Nanoseconds()
	fmt.Printf("n=%d\n", n)
	fmt.Printf("sum=%d\n", sum)
	fmt.Printf("elapsed_ns=%d\n", elapsed)
}

File: bench/micro/vec_amortized.js
// JS equivalent of vec_amortized.ap.

const n = 200_000;
const t0 = process.hrtime.bigint();

const v = [];
for (let i = 0; i < n; i++) {
    v.push(i);
}
let sum = 0;
for (let j = 0; j < n; j++) {
    sum = sum + v[j];
}

const elapsed = process.hrtime.bigint() - t0;
console.log(`n=${n}`);
console.log(`sum=${sum}`);
console.log(`elapsed_ns=${elapsed}`);
